src/Components/Credits/LoanAmortizationSimulation/FunctionForTest/numberOfBasePeriodsResult.ts

```diff
@@ -0,0 +1,39 @@
+type paymentPeriodOfInstallmentValue = {
+  [key: string]: number
+}
+
+const numberOfBasePeriodsResult = (
+  paymentPeriodOfInstallment: string,
+  durationInMonths: number
+) => {
+  if (
+    (paymentPeriodOfInstallment !== 'DurationInYears' &&
+      paymentPeriodOfInstallment !== 'DurationInHalfYears' &&
+      paymentPeriodOfInstallment !== 'DurationInQarters' &&
+      paymentPeriodOfInstallment !== 'DurationInMonths') ||
+    durationInMonths <= 0 ||
+    isNaN(durationInMonths)
```

Comment:
Move to variable

Reply:
Also you can create array of available `paymentPeriodOfInstallment ` and check if `array.includes(paymentPeriodOfInstallment)`. Apply in other places like that.